cloud/passportauth/src/user/UserRepository.ts

```diff
@@ -9,9 +7,10 @@ export interface UserRepository {
    * Retrieves a user object from a data source by login id (i.e. username, email)
    *
    * @param loginId - A unique login id used to identify the user
+   * @param cb - Callback
    * @returns {Promise<any>} - Returns the user data if user was found
    */
-  getUserByLogin(loginId: string): Promise<any>;
+  getUserByLogin(loginId: string, cb: (err: any, user?: any) => any): any;
```

Comment:
Proper pattern for cb:
```
cb: (err?: any, user?: any)
```

If we think that stacktraces are important here we can do:
```
cb: (err?: Error, user?: any)
```